cli/tests/pcluster/aws/dummy_aws_api.py

```diff
@@ -57,7 +55,14 @@ def max_network_interface_count(self):
         return self._max_network_interface_count
 
     def default_threads_per_core(self):
-        return self._default_threads_per_core
+        # There are more instance types, but for the simplicity of the mock,
+        # we consider only t2 as having one thread per core.
+        return 1 if self._instance_type.startswith("t2") else 2
+
+    def valid_threads_per_core(self):
+        # There are more instance types, but for the simplicity of the mock,
+        # we consider only t2 as having one thread per core.
```

Comment:
"we consider only t2 as having one thread per core."
probably you meant to consider only t2 has not reporting valid thread per core

Reply:
Yes, rephrasing in the next rev.